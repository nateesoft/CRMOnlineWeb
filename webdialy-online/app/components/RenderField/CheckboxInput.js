import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';

const InputCheckBox = ({ input, label }) => (
  <div>
    <FormControlLabel
      control={
        <Checkbox
          checked={input.value}
          onChange={input.onChange}
        />
      }
      label={label}
    />
  </div>
);

InputCheckBox.propTypes = {
  input: PropTypes.object,
  label: PropTypes.string,
  value: PropTypes.bool,
};

export default InputCheckBox;
