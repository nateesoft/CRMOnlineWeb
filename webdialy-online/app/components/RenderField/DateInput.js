/**
 *
 * RenderField
 *
 */

import React from 'react';
import TextField from '@material-ui/core/TextField';
import PropTypes from 'prop-types';

function DateInput({ input, label, required, meta: { touched, error } }) {
  return (
    <div>
      <TextField
        {...input}
        variant="outlined"
        margin="normal"
        type="date"
        fullWidth
        label={label}
        required={required}
        InputLabelProps={{
          shrink: true,
        }}
      />
      {touched && error && <span style={{ color: 'red' }}>{error}</span>}
    </div>
  );
}

DateInput.propTypes = {
  input: PropTypes.any,
  label: PropTypes.string,
  required: PropTypes.bool,
  meta: PropTypes.object,
};

export default DateInput;
