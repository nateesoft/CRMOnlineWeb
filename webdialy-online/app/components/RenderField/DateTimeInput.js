/**
 *
 * RenderField
 *
 */

import React from 'react';
import TextField from '@material-ui/core/TextField';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  errorLabel: {
    color: 'red',
  },
}));
function DateTimeInput({ input, label, required, meta: { touched, error } }) {
  const classes = useStyles();

  return (
    <>
      <TextField
        {...input}
        variant="outlined"
        margin="normal"
        type="date"
        fullWidth
        label={label}
        required={required}
        InputLabelProps={{
          shrink: true,
        }}
      />
      {touched && error && <span className={classes.errorLabel}>{error}</span>}
    </>
  );
}

DateTimeInput.propTypes = {
  input: PropTypes.any,
  label: PropTypes.object,
  required: PropTypes.bool,
  meta: PropTypes.object,
};

export default DateTimeInput;
