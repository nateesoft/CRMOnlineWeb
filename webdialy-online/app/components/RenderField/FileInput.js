/**
 *
 * FileInput
 *
 */

import React from 'react';

const FileInput = ({ input, meta: omitMeta, ...props }) => {
  return <input type="file" {...props.input} {...props} />;
};

export default FileInput;
