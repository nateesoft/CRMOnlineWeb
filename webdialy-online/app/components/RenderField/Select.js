/**
 *
 * RenderField
 *
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import FormControl from '@material-ui/core/FormControl';
import { makeStyles } from '@material-ui/core/styles';
import InputLabel from '@material-ui/core/InputLabel';

const useStyles = makeStyles(theme => ({
  formControl: {
    width: '100%',
    marginTop: theme.spacing(2),
  },
}));

function SelectField({ input, label, meta: { touched, error } }) {
  const classes = useStyles();
  const [prefix, setPrefix] = useState('');

  const handleChange = event => {
    setPrefix(event.target.value);
  };

  return (
    <FormControl variant="outlined" className={classes.formControl}>
      <InputLabel id="demo-simple-select-outlined-label">{label}</InputLabel>
      <Select
        {...input}
        labelId="demo-simple-select-outlined-label"
        native
        value={prefix}
        onChange={handleChange}
        label={label}
      >
        <option value="" />
        <option value="คุณ" key="คุณ">
          คุณ
        </option>
        <option value="นาย" key="นาย">
          นาย
        </option>
        <option value="นาง" key="นาง">
          นาง
        </option>
        <option value="นางสาว" key="นางสาว">
          นางสาว
        </option>
      </Select>
      {touched && error && <span style={{ color: 'red' }}>{error}</span>}
    </FormControl>
  );
}

SelectField.propTypes = {
  input: PropTypes.any,
  label: PropTypes.object,
  required: PropTypes.bool,
  autoFocus: PropTypes.bool,
  meta: PropTypes.object,
};

export default SelectField;
