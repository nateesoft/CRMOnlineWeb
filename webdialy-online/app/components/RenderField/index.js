/**
 *
 * RenderField
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';

function RenderField({
  input,
  label,
  type,
  required,
  autoFocus,
  disabled,
  meta: { touched, error },
}) {
  return (
    <div>
      <TextField
        {...input}
        variant="outlined"
        margin="normal"
        type={type}
        fullWidth
        label={label}
        required={required}
        autoFocus={autoFocus}
        disabled={disabled}
      />
      {touched && error && <span style={{ color: 'red' }}>{error}</span>}
    </div>
  );
}

RenderField.propTypes = {
  input: PropTypes.any,
  label: PropTypes.object,
  type: PropTypes.string,
  required: PropTypes.bool,
  autoFocus: PropTypes.bool,
  disabled: PropTypes.bool,
  meta: PropTypes.object,
};

export default RenderField;
