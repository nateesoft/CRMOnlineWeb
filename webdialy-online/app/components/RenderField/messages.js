/*
 * RenderField Messages
 *
 * This contains all the text for the RenderField component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.RenderField';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the RenderField component!',
  },
});
