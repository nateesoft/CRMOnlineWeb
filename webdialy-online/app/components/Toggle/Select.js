import styled from 'styled-components';

const Select = styled.select`
  line-height: 1em;
  background-color: transparent;
  border-style: none;
  margin-right: 10px;
  border: 1px solid white;
`;

export default Select;
