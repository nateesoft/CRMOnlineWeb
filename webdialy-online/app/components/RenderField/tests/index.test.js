import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { reduxForm, Field } from 'redux-form';

import history from 'utils/history';
import configureStore from 'configureStore';

import RenderField from '../index';
import CheckboxInput from '../CheckboxInput';
import DateInput from '../DateInput';
import DateTimeInput from '../DateTimeInput';
import FileInput from '../FileInput';
import Select from '../Select';

const initialState = {};

const store = configureStore(initialState, history);

describe('RenderField components', () => {
  it('RenderField snapshot', () => {
    const TestForm = () => (
      <form>
        <Field name="inputName" component={RenderField} type="text" margin="normal" value="test" />
      </form>
    );
    const ReduxSimpleForm = reduxForm({ form: 'newForm' })(TestForm);

    const component = renderer.create(
      <Provider store={store}>
        <ReduxSimpleForm />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('CheckboxInput snapshot', () => {
    const TestForm = () => (
      <form>
        <Field
          name="inputName"
          component={CheckboxInput}
          type="text"
          margin="normal"
          value="test"
        />
      </form>
    );
    const ReduxSimpleForm = reduxForm({ form: 'newForm' })(TestForm);

    const component = renderer.create(
      <Provider store={store}>
        <ReduxSimpleForm />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('DateInput snapshot', () => {
    const TestForm = () => (
      <form>
        <Field name="inputName" component={DateInput} type="text" margin="normal" value="test" />
      </form>
    );
    const ReduxSimpleForm = reduxForm({ form: 'newForm' })(TestForm);

    const component = renderer.create(
      <Provider store={store}>
        <ReduxSimpleForm />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('DateTimeInput snapshot', () => {
    const TestForm = () => (
      <form>
        <Field
          name="inputName"
          component={DateTimeInput}
          type="text"
          margin="normal"
          value="test"
        />
      </form>
    );
    const ReduxSimpleForm = reduxForm({ form: 'newForm' })(TestForm);

    const component = renderer.create(
      <Provider store={store}>
        <ReduxSimpleForm />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('FileInput snapshot', () => {
    const TestForm = () => (
      <form>
        <Field name="inputName" component={FileInput} type="text" margin="normal" value="test" />
      </form>
    );
    const ReduxSimpleForm = reduxForm({ form: 'newForm' })(TestForm);

    const component = renderer.create(
      <Provider store={store}>
        <ReduxSimpleForm />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Select snapshot', () => {
    const TestForm = () => (
      <form>
        <Field name="inputName" component={Select} type="text" margin="normal" value="test" />
      </form>
    );
    const ReduxSimpleForm = reduxForm({ form: 'newForm' })(TestForm);

    const component = renderer.create(
      <Provider store={store}>
        <ReduxSimpleForm />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
