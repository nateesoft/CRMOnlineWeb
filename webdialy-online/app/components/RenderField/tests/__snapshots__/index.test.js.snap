// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderField components CheckboxInput snapshot 1`] = `
<form>
  <label
    className="MuiFormControlLabel-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          className="PrivateSwitchBase-input-9"
          data-indeterminate={false}
          name="inputName"
          onChange={[Function]}
          type="checkbox"
          value=""
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
    <span
      className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    />
  </label>
</form>
`;

exports[`RenderField components DateInput snapshot 1`] = `
<form>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    onDragStart={[Function]}
    onDrop={[Function]}
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        name="inputName"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="date"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-3"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;

exports[`RenderField components DateTimeInput snapshot 1`] = `
<form>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    onDragStart={[Function]}
    onDrop={[Function]}
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        name="inputName"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="date"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-3"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;

exports[`RenderField components FileInput snapshot 1`] = `
<form>
  <input
    margin="normal"
    type="text"
  />
</form>
`;

exports[`RenderField components RenderField snapshot 1`] = `
<form>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    onDragStart={[Function]}
    onDrop={[Function]}
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        name="inputName"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-3"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;

exports[`RenderField components Select snapshot 1`] = `
<form>
  <div
    className="MuiFormControl-root makeStyles-formControl-12"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      id="demo-simple-select-outlined-label"
    />
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
    >
      <select
        aria-invalid={false}
        className="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        multiple={false}
        name="inputName"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        value=""
      >
        <option
          value=""
        />
        <option
          value="คุณ"
        >
          คุณ
        </option>
        <option
          value="นาย"
        >
          นาย
        </option>
        <option
          value="นาง"
        >
          นาง
        </option>
        <option
          value="นางสาว"
        >
          นางสาว
        </option>
      </select>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-3"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;
